File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/NumericAxis.swift
//
//  NumericAxis.swift
//  Swift Examples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

import Foundation

class NumericAxis: TKExamplesExampleViewController {
    
    let chart = TKChart()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        // >> chart-getting-started-swift
        chart.frame = self.view.bounds
        chart.autoresizingMask = UIViewAutoresizing(rawValue: UIViewAutoresizing.FlexibleWidth.rawValue | UIViewAutoresizing.FlexibleHeight.rawValue)
        self.view.addSubview(chart)
        // << chart-getting-started-swift
        
        // >> chart-getting-started-cus-swift
        chart.title.hidden = false
        chart.title.text = "This is a chart demo"
        chart.legend.hidden = false
        chart.allowAnimations = true
        // << chart-getting-started-cus-swift
        
        // >> chart-getting-started-data-swift
        var array = [TKChartDataPoint]()
        for i in 0..<12 {
            array.append(TKChartDataPoint(x: i, y: Int(arc4random() % 2000)))
        }
        // << chart-getting-started-data-swift
        
        // >> chart-getting-started-series-swift
        let series = TKChartLineSeries(items:array)
        series.selection = TKChartSeriesSelection.Series
        chart.addSeries(series)
        // << chart-getting-started-series-swift
        
        let xAxis = TKChartNumericAxis(minimum:0, andMaximum:11)
        xAxis.position = TKChartAxisPosition.Bottom
        xAxis.majorTickInterval = 1;
        chart.xAxis = xAxis
        
        // >> chart-axis-numeric-swift
        let yAxis = TKChartNumericAxis(minimum:0, andMaximum:2000)
        yAxis.majorTickInterval = 400
        yAxis.position = TKChartAxisPosition.Left
        yAxis.labelDisplayMode = TKChartNumericAxisLabelDisplayMode.Percentage
        chart.yAxis = yAxis
        // << chart-axis-numeric-swift
        
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}

File: Examples/Examples/Chart/AxisTypes/NumericAxis.cs
using System;
using System.Collections.Generic;

using Foundation;
using UIKit;

using TelerikUI;

namespace Examples
{
	[Register("NumericAxis")]
	public class NumericAxis: XamarinExampleViewController
	{
		public override void ViewDidLoad ()
		{
			base.ViewDidLoad ();

			// >> chart-getting-started-cs
			TKChart chart = new TKChart (this.View.Bounds);
			chart.AutoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight;
			this.View.AddSubview (chart);
			// << chart-getting-started-cs

			// >> chart-getting-started-cus-cs
			chart.Title.Hidden = false;
			chart.Title.Text = "This is a chart demo";
			chart.Legend.Hidden = false;
			chart.AllowAnimations = true;
			// << chart-getting-started-cus-cs

			// >> chart-getting-started-data-cs
			Random r = new Random ();
			List<TKChartDataPoint> list = new List<TKChartDataPoint> ();
			for (int i = 0; i < 12; i++) {
				list.Add (new TKChartDataPoint (new NSNumber (i), new NSNumber (r.Next () % 2000)));
			}
			// << chart-getting-started-data-cs

			// >> chart-getting-started-series-cs
			TKChartLineSeries series = new TKChartLineSeries (list.ToArray());

			series.Selection = TKChartSeriesSelection.Series;
			chart.AddSeries (series);

			// << chart-getting-started-series-cs

			TKChartNumericAxis xAxis = new TKChartNumericAxis ();
			xAxis.Range = new TKRange (new NSNumber (0), new NSNumber (11));
			xAxis.Position = TKChartAxisPosition.Bottom;
			xAxis.MajorTickInterval = 1;
			chart.XAxis = xAxis;

			// >> chart-axis-numeric-cs
			TKChartNumericAxis yAxis = new TKChartNumericAxis ();
			yAxis.Range = new TKRange (new NSNumber (0), new NSNumber (2000));
			yAxis.Position = TKChartAxisPosition.Left;
			yAxis.MajorTickInterval = 400;
			yAxis.LabelDisplayMode = TKChartNumericAxisLabelDisplayMode.Percentage;
			chart.YAxis = yAxis;
			// << chart-axis-numeric-cs

		}
	}
}

